### DISCUSSION TITLE
What is the most recommend way about using SWC in Node ?

### DISCUSSION THREAD

**Discussion by alexzhang1030** [swc-project/swc#6537] [Q&A] [answered]
I have tested the performance differences between `swc.transform` and `swc.transformSync` and `babel.transform`. Here is the [repo](https://github.com/alexzhang1030/swc-benchmark/tree/swc_vs_babel)

The result is 

```
swc:
  2 680 ops/s, ±27.15%    | slowest, 84.03% slower

swc sync:
  3 489 ops/s, ±36.64%    | 79.21% slower

babel:
  16 781 ops/s, ±2.44%   | fastest
```

So the fastest is babel. Is there something I missed? What is the most recommend way about using SWC in Node ?

Plz help me 🫠

**Comment by kdy1** [comment, marked as answer]
Babel does nothing by default. If you want to benchmark it, you have to pass compatibility passes to babel

**Reply by alexzhang1030** [reply]
Got it, thank you. BTW, is there any way to disable default compat of SWC, cause in some cases I only need plugin to control AST.

**Reply by alexzhang1030** [reply]
Hi kdy1 @kdy1 , could you answer this question? I will be very thankful ! 😄
